routes/profile.js

```diff
@@ -0,0 +1,82 @@
+require('../models/user.js');
+var auth = require("../modules/auth");
+var config = require('../config/config.js');
+var express = require('express');
+var mongoose = require("mongoose");
+var router = express.Router();
+var User = mongoose.model('User');
+
+router.get('/profile', auth.bounce, function(req, res) {
+  res.render('profile', {
+    userInfo: req.session.userInfo,
+    regExes: config.standards.htmlRegExes
+  });
+});
+
+router.post('/profile', auth.bounce, function(req, res) {
+  var infos = req.body;
+  // Sanity Checks
+  if (req.session.userInfo.cip.toString().toLocaleLowerCase().localeCompare(req.session.userInfo.cip.toString().toLocaleLowerCase()) === 0) {
+    // Prenom
+    if (req.session.userInfo.prenom.toString().toLocaleLowerCase().localeCompare(infos.prenom.toString().toLocaleLowerCase()) !== 0  && config.standards.regExes.name.test(infos.prenom)) {
+      req.session.userInfo.prenom = infos.prenom;
+    }
+    // Nom
+    if (req.session.userInfo.nom.toString().toLocaleLowerCase().localeCompare(infos.nom.toString().toLocaleLowerCase()) !== 0  && config.standards.regExes.name.test(infos.nom)) {
+      req.session.userInfo.nom = infos.nom;
+    }
+    // Email
+    if (req.session.userInfo.email.toString().toLocaleLowerCase().localeCompare(infos.email.toString().toLocaleLowerCase()) !== 0 && config.standards.regExes.email.test(infos.email)) {
+      req.session.userInfo.email = infos.email;
+    }
+    // Phone
+    if (typeof(req.session.userInfo.phone) === 'undefined') {
+      req.session.userInfo.phone='';
+    }
+    if (req.session.userInfo.phone.toString().toLocaleLowerCase().localeCompare(infos.phone.toString().toLocaleLowerCase()) !== 0 && config.standards.regExes.phone.test(infos.phone)) {
+      req.session.userInfo.phone = infos.phone;
+    }
+    // Concentration
+    if (req.session.userInfo.concentration.toString().toLocaleLowerCase().localeCompare(infos.concentration.toString().toLocaleLowerCase()) !== 0 && config.standards.regExes.concentration.test(infos.concentration)) {
+      req.session.userInfo.concentration = infos.concentration;
+    }
+    // Promotion
+    if (req.session.userInfo.promo.toString().toLocaleLowerCase().localeCompare(infos.promo.toString().toLocaleLowerCase()) !== 0 && config.standards.regExes.promo.test(infos.promo)) {
+      req.session.userInfo.promo = infos.promo;
+    }
+    // AGEG LDAP Username
+    if (typeof(req.session.userInfo.ageguname) === 'undefined') {
+      req.session.userInfo.ageguname='';
+    }
+    if (req.session.userInfo.ageguname.toString().toLocaleLowerCase().localeCompare(infos.username.toString().toLocaleLowerCase()) !== 0 && !config.standards.regExes.uname.test(infos.username)) {
+      req.session.userInfo.ageguname = infos.username;
+    }
```

Comment:
I really really don't like this code. Why don't you simply just update req.session.userInfo with the infos? We don't really care if it is the same or not!